com.unity.ml-agents/Runtime/SideChannels/EngineConfigurationChannel.cs

```diff
@@ -20,22 +30,34 @@ internal EngineConfigurationChannel()
         }
 
         /// <inheritdoc/>
-        public override void OnMessageReceived(IncomingMessage msg)
+        protected override void OnMessageReceived(IncomingMessage msg)
         {
-            var width = msg.ReadInt32();
-            var height = msg.ReadInt32();
-            var qualityLevel = msg.ReadInt32();
-            var timeScale = msg.ReadFloat32();
-            var targetFrameRate = msg.ReadInt32();
-            var captureFrameRate = msg.ReadInt32();
-
-            timeScale = Mathf.Clamp(timeScale, 1, 100);
-
-            Screen.SetResolution(width, height, false);
-            QualitySettings.SetQualityLevel(qualityLevel, true);
-            Time.timeScale = timeScale;
-            Time.captureFramerate = captureFrameRate;
-            Application.targetFrameRate = targetFrameRate;
+            var messageType = (ConfigurationType)msg.ReadInt32();
+            switch (messageType)
+            {
+                case ConfigurationType.Screen:
+                    var width = msg.ReadInt32();
+                    var height = msg.ReadInt32();
+                    Screen.SetResolution(width, height, false);
+                    break;
+                case ConfigurationType.QualityLevel:
+                    var qualityLevel = msg.ReadInt32();
+                    QualitySettings.SetQualityLevel(qualityLevel, true);
+                    break;
+                case ConfigurationType.TimeScale:
+                    var timeScale = msg.ReadFloat32();
+                    timeScale = Mathf.Clamp(timeScale, 1, 100);
+                    Time.timeScale = timeScale;
+                    break;
+                case ConfigurationType.TargetFrameRate:
+                    var targetFrameRate = msg.ReadInt32();
+                    Application.targetFrameRate = targetFrameRate;
+                    break;
+                case ConfigurationType.CaptureFrameRate:
+                    var captureFrameRate = msg.ReadInt32();
+                    Time.captureFramerate = captureFrameRate;
+                    break;
+            }
```

Comment:
I left a comment in the other PR in EnvironmentParametersChannel.OnMessageReceived() - we should be consistent with how we handle unknown values. I don't think we should throw (otherwise it's not forwards compatible) but we should either do nothing (like here) or warn.

Reply:
I can send a warning.